src/server/pfs/fuse/server.go

```diff
@@ -447,6 +562,90 @@ func Server(c *client.APIClient, sopts *ServerOptions) error {
 	return errors.EnsureStack(srv.ListenAndServe())
 }
 
+func getClusterStatus(c *client.APIClient) (map[string]string, error) {
+	var clusterStatus string
+	err := c.Health()
+	if err != nil {
+		clusterStatus = "INVALID"
+	} else {
+		authActive, err := c.IsAuthActive()
+		if err != nil {
+			return nil, err
+		}
+
+		if authActive {
+			clusterStatus = "AUTH_ENABLED"
+		} else {
+			clusterStatus = "AUTH_DISABLED"
+		}
+	}
+
+	pachdAddress := c.GetAddress().Qualified()
+
+	return map[string]string{"cluster_status": clusterStatus, "pachd_address": pachdAddress}, nil
+}
+
+func updateClientEndpoint(reqPachdAddress string, c *client.APIClient) (map[string]string, error) {
+	cfg, err := config.Read(false, false)
+	if err != nil {
+		return nil, err
+	}
+	contextName, _, err := cfg.ActiveContext(true)
+	if err != nil {
+		return nil, err
+	}
+	pachdAddress, err := grpcutil.ParsePachdAddress(reqPachdAddress)
+	if err != nil {
+		return nil, errors.WithStack(fmt.Errorf("either empty or poorly formatted cluster endpoint"))
+	}
+
+	// Check if same pachd address as current client
+	var clusterStatus map[string]string
+	if reflect.DeepEqual(pachdAddress, c.GetAddress()) {
+		clusterStatus, err = getClusterStatus(c)
+		if err != nil {
+			return nil, err
+		}
+		logrus.Infof("New endpoint is same as current endpoint: %s\n", clusterStatus["pachd_address"])
+	} else {
+		// Check if new cluster endpoint is valid
+		newClient, err := client.NewFromPachdAddress(pachdAddress)
+		if err != nil {
+			clusterStatus = map[string]string{"cluster_status": "INVALID", "pachd_address": pachdAddress.Qualified()}
+		} else {
+			defer newClient.Close()
+			clusterStatus, err = getClusterStatus(newClient)
+			if err != nil {
+				return nil, err
+			}
+			if clusterStatus["cluster_address"] != "INVALID" {
+				logrus.Infof("Updating pachd address from %s to %s\n", c.GetAddress().Qualified(), pachdAddress.Qualified())
+				cfg.V2.Contexts[contextName] = &config.Context{PachdAddress: pachdAddress.Qualified()}
+				err = cfg.Write()
+				if err != nil {
+					return nil, err
+				}
+
+				newClientPtr, err := client.NewOnUserMachine("fuse")
+				if err != nil {
+					return nil, err
+				}
+				*c = *(newClientPtr)
+			}
+		}
+	}
+
+	return clusterStatus, nil
+}
+
+func jsonMarshal(t interface{}) ([]byte, error) {
```

Comment:
Just noting here, we should use pachyderm's serde package (which has extra handling for protobufs and a few other tweaks) here, for consistency with the rest of pachyderm/pachyderm

Reply:
When returning the auth URL login, the default JSON encoding was replacing characters that invalidated the URL. Hence I created this function that disabled escaping certain HTML characters. 

I took a look at the `EncodeTransform` function in the serde package. Super useful that it allows transformations on the data to be returned. Correct me if I'm wrong but that wouldn't help in this case right since my issue isn't with the data itself but rather a setting of the encoder, which `EncodeTransform` doesn't allow me to modify.